arch/xtensa/src/esp32s3/esp32s3_serial.c

```diff
@@ -1124,7 +1108,7 @@ int up_putc(int ch)
       up_lowputc('\r');
     }
 
-  up_lowputc(ch);
+  up_lowputc((char)ch);
```

Comment:
I do not think that this is needed

Reply:
This is for silencing clang-tidy's *bugprone-narrowing-conversions* warning:
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-narrowing-conversions.html